packages/core/src/fhir-to-cda/cda-templates/components/medications.ts

```diff
@@ -0,0 +1,198 @@
+    const refResource = ref ? findResourceInBundle(fhirBundle, ref) : undefined;
+    const medication = isMedication(refResource) ? refResource : undefined;
+    return new AugmentedMedicationStatement(statement, medicationsSectionName, medication);
+  });
+
+  const { trs, entries } = createTableRowsAndEntries(
+    augmentedMedStatements,
+    createTableRowsFromMedicationStatements,
+    createEntryFromStatement
+  );
+
+  const table = initiateSectionTable(medicationsSectionName, tableHeaders, trs);
+
+  const medicationsSection = {
+    component: {
+      section: {
+        templateId: buildInstanceIdentifier({
+          root: "2.16.840.1.113883.10.20.22.2.1.1",
+        }),
+        code: buildCodeCe({
+          code: "10160-0",
+          codeSystem: loincCodeSystem,
+          codeSystemName: loincSystemName,
+          displayName: "History of Medication use Narrative",
+        }),
+        title: "MEDICATIONS",
+        text: { table },
+        entry: entries,
+      },
+    },
+  };
+  return medicationsSection;
+}
+
+function isMedicationStatement(resource: Resource | undefined): resource is MedicationStatement {
+  return resource?.resourceType === "MedicationStatement";
+}
+
+function isMedication(resource: Resource | undefined): resource is Medication {
+  return resource?.resourceType === "Medication";
+}
+
+function createTableRowsFromMedicationStatements(
+  statement: AugmentedMedicationStatement,
+  medicationsPrefix: string
+): ObservationTableRow[] {
+  const trs: ObservationTableRow[] = [];
+  const tableRow = createTableRowFromObservation(statement, medicationsPrefix);
+  if (tableRow) trs.push(tableRow);
+  return trs;
+}
+
+function createTableRowFromObservation(
+  statement: AugmentedMedicationStatement,
+  referenceId: string
+): ObservationTableRow | undefined {
+  const period = {
+    start: statement.resource?.effectivePeriod?.start,
+    end: statement?.resource?.effectivePeriod?.end,
+  };
+  const medicationName = statement.medication?.code?.text;
+  if (!medicationName) return;
+
+  return {
+    tr: {
+      [_idAttribute]: referenceId,
+      ["td"]: [
+        {
+          [_inlineTextAttribute]: medicationName,
+        },
+        {
+          [_inlineTextAttribute]: getDosageFromMedicationStatement(statement.resource),
+        },
+        {
+          [_inlineTextAttribute]: getFrequencyFromMedicationStatement(statement.resource),
+        },
+        {
+          [_inlineTextAttribute]: formatDateToHumanReadableFormat(period.start) ?? NOT_SPECIFIED,
+        },
+        {
+          [_inlineTextAttribute]: formatDateToHumanReadableFormat(period.end) ?? NOT_SPECIFIED,
+        },
+        {
+          [_inlineTextAttribute]: statement.resource.reasonCode?.[0]?.text ?? NOT_SPECIFIED,
+        },
+      ],
+    },
+  };
+}
+
+function getDosageFromMedicationStatement(statement: MedicationStatement): string {
+  const dosageValue = statement.dosage?.[0]?.doseAndRate?.[0]?.doseQuantity?.value;
```

Comment:
always in the 0th index?

Reply:
Yea, that should be sufficient. I haven't seen examples where there would be multiple elements in the `doseAndRate` array for a medication. And this is a pattern that's seen across different resources/attributes